sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -164,11 +164,42 @@ case class Join(
         left.output.map(_.withNullability(true)) ++ right.output
       case FullOuter =>
         left.output.map(_.withNullability(true)) ++ right.output.map(_.withNullability(true))
+      case NaturalJoin(jt) =>
+        outerProjectList(jt).map(_.toAttribute)
       case _ =>
         left.output ++ right.output
     }
   }
 
+  def outerProjectList(jt: JoinType): Seq[NamedExpression] = {
```

Comment:
there are a lot of duplicated code here with output (although they are slightly different). can we rewrite them to make it simpler?